enhancements/bootimages.md

```diff
@@ -0,0 +1,163 @@
+ 
+### Goal 3: Better integrate bootimage management into disconnected install paths
+
+OpenShift 4 comes as a "release image" which is a *container* image - bootimages do not naturally fit into that, and currently the installer has some ad-hoc support for dealing with bootimages.
+
+A goal here is to make bootimage management more of a first class operation, something like:
+
+```
+$ oc adm release generate-bootimage quay.io/openshift/release:4.3 vmware
+```
+
+Rather than having an administrator [manually download a bootimage](https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.3/latest/) corresponding to the release version, this command would output a bootimage for the chosen platform/media.
+
+Similarly:
+
+```
+$ oc adm release generate-bootimage quay.io/openshift/release:4.3 aws
+```
+
+would output data similar to the [aws rhcos.json data](https://github.com/openshift/installer/blob/2055609f95b19322ee6cfdd0bea73399297c4a3e/data/data/rhcos.json#L2) so that AWS UPI installations could use it programatically.
+
+Other output proposals
+
+- `metal`, `openstack`, `qemu`: What's available today for 4.3
+- `aws-snapshot` to handle [private AWS regions](https://github.com/openshift/os/blob/411d1f5943ea23f2de7e4f7a04ab0bb185fd9586/FAQ.md#q-how-do-i-get-rhcos-in-a-private-ec2-region).
+- `baremetal-iso` generates a "Live" ISO as is used in Fedora CoreOS and points people at how to do [Ignition injection](https://github.com/coreos/coreos-installer/blob/3652e6a767bad593b1b898f239e41bc11a83ab8f/src/iso.rs#L28)
+- `baremetal-pxe` outputs the split kernel/initramfs suitable for PXE
+
+### Non-Goals
+
+#### Replacing the default in-place update path
+
+In-place updates as [managed by the MCO](https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md) today works fairly seamlessly.  We can't require that everyone fully reprovision a machine in order to do in-place updates - that makes updates *much* more expensive.  It implies re-downloading all container images, etc.
+
+#### Exposing a general purpose "OS build" tool
+
+This proposal includes shipping a subset of https://github.com/coreos/coreos-assembler/ - but that's not the same as supporting custom "CoreOS-style" builds for non-OpenShift use cases.
+
+## Proposal phase 1: bootimage.json in release image
+
+First, the RHCOS build process is changed to inject the current coreos-assembler `meta.json` output for the build into `machine-os-content`.  This aims to move the "source of truth" for cluster bootimages into the release image.  Nothing would use this to start - however, as soon as we switch to a machine API provisioned control plane, having that process consume this data would be a natural next step.
```

Comment:
Could you have some details about what this means? 

I've read through what I can find and I don't understand how machine-os-content is produced, there's no source location in the release image. 

As a user, how would I consume this? How would I find out which is the image URL for say the OpenStack image? Does the meta.json become an annotation on the image or something?


```json
        {
          "name": "machine-os-content",
          "annotations": {
            "io.openshift.build.commit.id": "",
            "io.openshift.build.commit.ref": "",
            "io.openshift.build.source-location": "",
            "io.openshift.build.version-display-names": "machine-os=Red Hat Enterprise Linux CoreOS",
            "io.openshift.build.versions": "machine-os=45.81.202003131108-0"
          }
```

Reply:
None of this exists yet - it's a proposal.

But this is the same question as https://github.com/openshift/enhancements/pull/201#discussion_r375546207 right?

Reply:
>  I don't understand how machine-os-content is produced

Via https://github.com/coreos/coreos-assembler and https://gitlab.cee.redhat.com/coreos/redhat-coreos/

Reply:
Got it thank you - I was looking for details about how this should be implemented to see if we could already do part of it. 

The problem for us is that baremetal (and openstack) IPI both need to know where the RHCOS images live. Today we do awful hacks with `openstack-install version` and GitHub to fetch the data that don't work in CI. There, `openshift-install version` reports a sha that doesn't exist due to the rebasing strategy CI uses. We ended up with a temporary workaround and just copy rhcos.json into the installer container.

I'd be interested to see if we could get a better approach to make the data accessible to end users (this `--info` flag @abhinavdahiya mentioned)